v2/internal/testcommon/samples_tester.go

```diff
@@ -0,0 +1,226 @@
+// "runtime.notRegisteredErr" error from the Scheme.
+var exclusions = [...]string{"batchaccountpool", "snapshot",
+	"sqldatabasecontainerthroughputsetting", "sqldatabasethroughputsetting",
+	"webtest", "storageaccountsqueueservice",
+	// excluding dbformysql/user as is not a resource
+	"user",
+}
+
+type SamplesTester struct {
+	tc            *KubePerTestContext
+	decoder       runtime.Decoder
+	UseRandomName bool
+}
+
+type SampleObject struct {
+	SamplesMap   map[string]*linkedhashmap.Map
+	childObjects map[string]genruntime.ARMMetaObject
+}
+
+func NewSampleObject() SampleObject {
+	return SampleObject{
+		SamplesMap:   make(map[string]*linkedhashmap.Map),
+		childObjects: make(map[string]genruntime.ARMMetaObject),
+	}
+}
+
+func NewSamplesTester(tc *KubePerTestContext, useRandomName bool) *SamplesTester {
+	return &SamplesTester{
+		tc:            tc,
+		decoder:       serializer.NewCodecFactory(tc.GetScheme()).UniversalDecoder(),
+		UseRandomName: useRandomName,
+	}
+}
+
+func (t SamplesTester) LoadSamples(rg *resources.ResourceGroup, group string) (*SampleObject, *SampleObject) {
+
+	samples := NewSampleObject()
+	depSamples := NewSampleObject()
+
+	groupPath := getGroupPath(group)
+
+	err := filepath.Walk(groupPath,
+		func(filePath string, info os.FileInfo, err error) error {
+
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() && !isExclusion(filePath) {
+				//if !info.IsDir() && !isExclusion(filePath) {
+
+				sample := t.getObjectFromFile(filePath)
+
+				sample.SetNamespace(t.tc.Namespace)
+
+				if !strings.Contains(filePath, RefsPackage) {
+					if t.UseRandomName {
+						sample.SetName(t.tc.NoSpaceNamer.GenerateName(""))
+					}
+					handleObject(sample, rg, &samples)
+				} else {
+					// We don't change name for dependencies as they have fixed refs which we can't access inside the object
+					// need to make sure we have right names for the refs in samples
+					handleObject(sample, rg, &depSamples)
+				}
+			}
+
+			return nil
+		})
+
+	t.tc.Expect(err).To(BeNil())
+
+	// We process the child objects once we have all the parent objects in the tree already
+	handleChildObjects(t.tc, &samples)
+	handleChildObjects(t.tc, &depSamples)
+
+	return &samples, &depSamples
+}
+
+func (t SamplesTester) getObjectFromFile(path string) genruntime.ARMMetaObject {
+	jsonMap := make(map[string]interface{})
+
+	byteData, err := ioutil.ReadFile(path)
+	t.tc.Expect(err).To(BeNil())
+
+	jsonBytes, err := yaml.ToJSON(byteData)
+	t.tc.Expect(err).To(BeNil())
+
+	err = json.Unmarshal(jsonBytes, &jsonMap)
+	t.tc.Expect(err).To(BeNil())
+
+	// We need unstructured object here to fetch the correct GVK for the object
+	unstructuredObj := unstructured.Unstructured{Object: jsonMap}
+
+	obj, err := genruntime.NewObjectFromExemplar(&unstructuredObj, t.tc.GetScheme())
+	t.tc.Expect(err).To(BeNil())
+
+	byteData, err = json.Marshal(unstructuredObj.Object)
+	t.tc.Expect(err).To(BeNil())
+
+	err = runtime.DecodeInto(t.decoder, byteData, obj)
```

Comment:
Why do you need to remarshal from `unstructuredObj` back to `byteData` instead of just reusing `byteData` since it's already populated, presumably with the same content?

Reply:
True! I had some mangling around with the byte data before parsing to set the owners. For which I found solution using `reflect` and have missed out on removing this. My bad